tpm2/kdf.go

```diff
@@ -0,0 +1,57 @@
+package tpm2
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"hash"
+)
+
+// KDFa implements TPM 2.0's default key derivation function, as defined in
+// section 11.4.9.2 of the TPM revision 2 specification part 1.
+// The key & label parameters must not be zero length, but contextU &
+// contextV may be.
+func KDFa(hashAlg Algorithm, key []byte, label string, contextU, contextV []byte, bits int) ([]byte, error) {
```

Comment:
Are contextU and contextV expected to be 0-terminated strings? Should we verify that? Or should they be strings instead of []byte?

Reply:
contextU/contextV are a bunch of different things depending on the command/context its being used for (we cant enforce anything here). The only one that is guaranteed to be a 0-terminated string is label, and we handle that already by accepting the string type.